components/ui/pathogen-map/pathogen-map.tsx

```diff
@@ -122,12 +125,17 @@ export function PathogenMap<
   const onRender = (event: mapboxgl.MapboxEvent) => {
     const map = event.target;
     if (map) {
-      const features = map.querySourceFeatures(sourceId);
+      const features = map.querySourceFeatures(sourceId) as any as GeoJSON.Feature<
+        GeoJSON.Geometry,
+        { cluster: boolean, cluster_id: string } & Record<TClusterPropertyKey, number>
+      >[];
```

Comment:
:vomiting_face: 
Tried my best but I think the cast here is inevitable, the type returned by this function and the type my function uses are just too incompatible but this is the correct typing. Maybe I can figure out some kind of complicated type validation for this later but this will have to do for now :/